src/main/kotlin/no/nav/brukerdialog/ytelse/opplæringspenger/api/OpplæringspengerSøknadController.kt

```diff
@@ -0,0 +1,70 @@
+package no.nav.brukerdialog.ytelse.opplæringspenger.api
+
+import kotlinx.coroutines.runBlocking
+import no.nav.brukerdialog.common.MetaInfo
+import no.nav.brukerdialog.common.formaterStatuslogging
+import no.nav.brukerdialog.config.Issuers
+import no.nav.brukerdialog.domenetjenester.innsending.InnsendingCache
+import no.nav.brukerdialog.domenetjenester.innsending.InnsendingService
+import no.nav.brukerdialog.integrasjon.k9selvbetjeningoppslag.BarnService
+import no.nav.brukerdialog.metrikk.MetrikkService
+import no.nav.brukerdialog.utils.MDCUtil
+import no.nav.brukerdialog.utils.NavHeaders
+import no.nav.brukerdialog.utils.TokenUtils.personIdent
+import no.nav.brukerdialog.ytelse.opplæringspenger.api.domene.OpplæringspengerSøknad
+import no.nav.security.token.support.core.api.ProtectedWithClaims
+import no.nav.security.token.support.core.api.RequiredIssuers
+import no.nav.security.token.support.spring.SpringTokenValidationContextHolder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.web.ErrorResponseException
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/opplaringspenger")
+@RequiredIssuers(
+    ProtectedWithClaims(issuer = Issuers.TOKEN_X, claimMap = ["acr=Level4"])
+)
+class OpplæringspengerSøknadController(
+    private val innsendingService: InnsendingService,
+    private val innsendingCache: InnsendingCache,
+    private val barnService: BarnService,
+    private val springTokenValidationContextHolder: SpringTokenValidationContextHolder,
+    private val metrikkService: MetrikkService,
+) {
+    private companion object {
+        private val logger: Logger = LoggerFactory.getLogger(OpplæringspengerSøknadController::class.java)
+    }
+
+    @PostMapping("/innsending", consumes = [MediaType.APPLICATION_JSON_VALUE])
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    fun innsending(
+        @RequestHeader(NavHeaders.BRUKERDIALOG_YTELSE) ytelse: String,
+        @RequestHeader(NavHeaders.BRUKERDIALOG_GIT_SHA) gitSha: String,
+        @Value("\${ENABLE_OPPLAERINGSPENGER:false}") enabled: Boolean? = null,
```

Comment:
Burde denne ligge på `class` i stedet for i `fun`? 🤔

Reply:
Den fungerer der også, men tåker litt til endepunktet.

Ville ha flyttet til den konstruktøren istedenfor. 
Ser det er samme på `ENABLE_UNDOMSYTELSE`.